### PR TITLE
adding scale dependent styling

### PR DESCRIPTION
**Pull request by davidmsibley**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /*/target/
 /*/src/main/webapp/META-INF/
 */nbactions.xml
-!/glri-afinch-web/src/main/webapp/META-INF/
```
Comment (davidmsibley) [inline .gitignore]: I realize now, star is a little overkill, but it works.  However, why was it ! in the first place? typo?

File: glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js [hunk 2]
```diff
@@ -9,58 +9,115 @@
...
-        }),
-        'select': new OpenLayers.Style({
-            'pointRadius': 2,
-            'fillColor': '#609428',
-            'fillOpacity': 0.4,
-            'strokeColor': '#ffffff',
-            'strokeOpacity': 1,
-            'strokeWidth': 3
-        }),
-        renderers: ['DeclusterCanvas']
```
Comment (davidmsibley) [inline glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js]: As far as I could tell DeclusterCanvas isn't in the project.  It's using the SVG renderer.  Is that a mistake?  I took decluster canvas out to be explicit.

File: glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js [hunk 3]
```diff
@@ -9,58 +9,115 @@
...
+	styleMap: new OpenLayers.StyleMap({
+		'default': new OpenLayers.Style({
+			'fillColor': '#558027',
+			'fillOpacity': 0.4,
+			'strokeColor': '#558027',
+			'strokeOpacity': 1,
+			'strokeWidth': 1,
+			'graphicName': 'triangle'
+		}, {
+			rules: [
```
Comment (davidmsibley) [inline glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js]: I'd love to pull this out into a reasonably scoped var.  Just say the word.

**Comment by davidmsibley** [comment]
assigning @ee-usgs, also FYI @cschroed-usgs if you're interested

File: glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js (hunk 2 above)
Comment (ee-usgs) [inline glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js]: I’m confused by the Decluster thing too, but, there is a lot going on here.
This is Tom’s ‘Render in the UI based on data encoded in image tiles’ framework.  I was really hoping to rip that feature out for the gage stations and eventually for the flow lines as well.

Are you able to verify that this works?  We could merge and try a deployment if needed.  For some reason, testing locally doesn’t seem to work - I just get little green boxes for the gage stations.  I *think* that was the case before as well - for some reason local testing of layers that use Tom’s fancy rendering were busted and I haven’t had a task that required me to figure it out.

Comment (davidmsibley) [inline glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js]: I didn't rip anything out, the gage locations layer is just a vector layer, that has a normal renderer

Comment (cschroed-usgs) [inline glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js]: > I was really hoping to rip that feature out for the gage stations and eventually for the flow lines as well.

When I was working on the project, we were using Tom's canvas pipeline feature to implement a requirement -- zoom-level-sensitive thinning of the reach network. The rationale is that when you're zoomed way out, it doesn't make sense to view Farmer Brown's creek; you want to see the Muskegon River. Zoomed out = less detailed reach networks. Zoomed in = detailed reach networks. Openlayers and Geoserver don't support that out-of-the-box, hence the canvas rendering pipeline. If you rip it out, make sure that you communicate with stakeholders to ensure that zoom-sensitive reach network thinning is no longer a requirement.

Comment (jlucido2) [inline glri-afinch-web/src/main/webapp/js/openlayers/extension/Layer/GageFeature.js]: I want to be clear that 'zoom-sensitive reach network thinning' IS still a requirement.

**Pull request merged**